File: LeetCode/2149-remove-colored-pieces-if-both-neighbors-are-the-same-color/2149-remove-colored-pieces-if-both-neighbors-are-the-same-color.cpp
class Solution {
public:
    bool winnerOfGame(string colors) {
        int n = colors.length(), c1 = 0, c2 = 0;
        if (n < 3) {
            return false;
        }
        for (int i = 1; i < n - 1; i++) {
            if ((colors[i] == colors[i - 1]) && (colors[i] == colors[i + 1]) && (colors[i] == 'A')) {
                c1 += 1;
            }
            if ((colors[i] == colors[i - 1]) && (colors[i] == colors[i + 1]) && (colors[i] == 'B')) {
                c2 += 1;
            }
        }
        return c1 > c2;
    }
};

File: LeetCode/2149-remove-colored-pieces-if-both-neighbors-are-the-same-color/2149-remove-colored-pieces-if-both-neighbors-are-the-same-color.java
class Solution {
    public boolean winnerOfGame(String colors) {
        int n = colors.length(), c1=0, c2=0;
        for(int i=1;i<n-1;i++)
        {
            if(colors.charAt(i) == colors.charAt(i-1) && colors.charAt(i) == colors.charAt(i+1))
            {
                if(colors.charAt(i)=='A')
                {
                    c1 += 1;
                }
                if(colors.charAt(i)=='B')
                {
                    c2 += 1;
                }
            }
        }
        return c1 > c2;
    }
}